tap-gui/configurator/api-docs.md

```diff
@@ -0,0 +1,602 @@
+
+## Class BackendPluginSurface
+
+Add plugins to the Backstage backend.
+
+### Constructors
+
+#### constructor()
+
+Constructs a new instance of the `BackendPluginSurface` class
+
+### Methods
+
+#### addPlugin(plugin: Plugin): void
+
+Register a named plugin to handle backend requests at some subpath.
+
+Plugins must have unique names. The first plugin registered with a given name wins.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  plugin | Plugin |  |
+
+# Package @vmware-tanzu/core-frontend
+
+## Class ApiSurface
+
+Add Backstage ApiFactories.
+
+### Methods
+
+#### add(factory: AnyApiFactory): void
+
+Register a new Backstage ApiFactory.
+
+Factories must specify an API with a unique ID. The last factory registered with a given ID wins.
+
+Factories that specify APIs with IDs on the ignored list will not be registered.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  factory | AnyApiFactory |  |
+
+## Class AppComponentSurface
+
+Substitute replaceable core components of the Backstage app.
+
+### Constructors
+
+#### constructor()
+
+Constructs a new instance of the `AppComponentSurface` class
+
+### Methods
+
+#### add&lt;K extends keyof AppComponents&gt;(key: K, component: AppComponents\[K\]): void
+
+Substitute a given Backstage AppComponent.
+
+AppComponents must be unique by key. The last AppComponent registered with a given key wins.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  key | K | Unique key for the AppComponent. |
+|  component | AppComponents\[K\] | Custom AppComponent to substitute for Backstage's default. |
+
+## Class AppPluginSurface
+
+Register additional frontend plugins with the Backstage app.
+
+### Constructors
+
+#### constructor()
+
+Constructs a new instance of the `AppPluginSurface` class
+
+### Methods
+
+#### add(plugin: BackstagePlugin): void
+
+Register a frontend BackstagePlugin.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  plugin | BackstagePlugin |  |
+
+## Class AppRouteSurface
+
+Manipulate routes in the frontend Backstage app.
+
+### Constructors
+
+#### constructor()
+
+Constructs a new instance of the `AppRouteSurface` class
+
+### Methods
+
+#### add(route: ReactElement): void
+
+Register a react-router Route.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  route | ReactElement | Route to register. Intended to be a react-router\#Route or a subclass of it. |
+
+#### addRouteBinder(routeBinder: RouteBinder): void
+
+Bind external routes.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  routeBinder | RouteBinder |  |
+
+## Class BannerSurface
+
+Add page-level banners to the Backstage app.
+
+### Constructors
+
+#### constructor()
+
+Constructs a new instance of the `BannerSurface` class
+
+### Methods
+
+#### add(banner: ReactElement): void
+
+Add a new page-level banner.
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  banner | ReactElement |  |
+
+## Class EntityPageSurface
```

Comment:
is this a duplicate?

Reply:
Yeah, it dupes the one in `tpb-catalog`. I just need to mechanically verify that the one in the plugin is the active one — I'm 99% certain but need to do the due diligence — and then remove this one. Good look, thanks.